### PR TITLE
FIX-#3985: Update pyarrow in environment files

### LINKED ISSUE #3985: Update pyarrow in environment files
When creating an env from enviroment.yml pyarrow 3.0.0 is installed. At the same time, when installing modin-ray and modin-omnisci in the same env pyarrow 5.0.0 is installed. So we should update pyarrow in requirement-dev.txt and enviroment.yml files to check correctness tests pass with a newer pyarrow version.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3985  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ channels:
 dependencies:
   - pandas==1.3.5
   - numpy>=1.16.5
-  - pyarrow>=1.0.0
+  - pyarrow>=6.*
```
Comment (vnlitvinov) [inline environment-dev.yml:7]: Don't use relative comparison and asterisk at the same time, conda gives a warning and discards the asterisk anyway.

Use `pyarrow>=6.0` to achieve the same effect.

**Comment by YarShev** [comment]
Pyarrow should be updated in setup.py too.

File: environment-dev.yml (hunk 1 above)
Comment (YarShev) [reply environment-dev.yml]: fixed

**Comment by YarShev** [comment]
libthrift<0.14.1 is a culprit of the conflict with a newer pyarrow so removed it to allow pyarrow itself to install libthrift.

File: setup.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
     long_description = fh.read()
 
 dask_deps = ["dask>=2.22.0", "distributed>=2.22.0"]
-ray_deps = ["ray[default]>=1.4.0", "pyarrow>=1.0"]
+ray_deps = ["ray[default]>=1.4.0", "pyarrow>=6.0"]
```
Comment (YarShev) [inline setup.py:8]: An option for pinning pyarrow would be >= 4.0.1 like Ray requires it as extras ([link])(https://github.com/ray-project/ray/blob/master/python/setup.py#L197). However, our requirements on pyarrow version for modin-ray and modin-omnisci should be aligned. Any thoughts on what a better option would be?

**Comment by Garra1980** [comment]
I'd say >= 4.0.1 is fine here

**Review by devin-petersohn — APPROVED** [review]
LGTM!

**Review by vnlitvinov — APPROVED** [review]

**Pull request merged**